pkg/controller/certificates/sync.go

```diff
@@ -114,20 +124,51 @@ func (c *Controller) Sync(ctx context.Context, crt *v1alpha1.Certificate) (err e
 		return err
 	}
 
+	// update certificate expiry metric
+	defer c.updateCertificateExpiryMetric(crt)
+
 	// as there is an existing certificate, or we may create one below, we will
 	// run scheduleRenewal to schedule a renewal if required at the end of
 	// execution.
 	defer c.scheduleRenewal(crt)
 
 	crtCopy := crt.DeepCopy()
 
+	// get issuer details for metrics
+	issuerName := issuerObj.GetObjectMeta().Name
+	var issuerKind string
+	switch crt.Spec.IssuerRef.Kind {
+	case "", v1alpha1.IssuerKind:
+		issuerKind = v1alpha1.IssuerKind
+	case v1alpha1.ClusterIssuerKind:
+		issuerKind = v1alpha1.ClusterIssuerKind
+	default:
+		return fmt.Errorf(`invalid value %q for certificate issuer kind. Must be empty, %q or %q`, crt.Spec.IssuerRef.Kind, v1alpha1.IssuerKind, v1alpha1.ClusterIssuerKind)
+	}
+
 	// if the certificate was not found, or the certificate data is invalid, we
 	// should issue a new certificate.
 	// if the certificate is valid for a list of domains other than those
 	// listed in the certificate spec, we should re-issue the certificate.
 	if k8sErrors.IsNotFound(err) || errors.IsInvalidData(err) ||
 		expectedCN != cert.Subject.CommonName || !util.EqualUnsorted(cert.DNSNames, expectedDNSNames) {
+
 		err := c.issue(ctx, i, crtCopy)
+
+		// count result
+		result := resultSuccess
+		if err != nil {
+			result = resultFailure
+		}
+		c.metrics.IncCertificateRequestCount(
+			crt.Name,
+			crt.Namespace,
+			issuerName,
+			issuerKind,
+			issuerType,
+			operationIssue,
+			result)
```

Comment:
I'd like to see this call be a simple `c.metrics.HandleCertificateIssue(crt, err)`